src/main/java/io/cryostat/agent/MainModule.java

```diff
@@ -302,13 +312,17 @@ public static SSLContext provideClientSslContext(
                     clearBuffer(keystorePass);
                     clearBuffer(keyPass);
                 }
-            } else if (clientAuthCertPath.isPresent() || clientAuthKeyPath.isPresent()) {
+            } else if (clientAuthCertPath.isPresent()
+                    || clientAuthKeyPath.isPresent()
+                    || tlsEnabled) {
                 throw new IllegalArgumentException(
                         String.format(
                                 "To use TLS client authentication, both the certificate (%s) and"
-                                        + " private key (%s) properties must be set.",
+                                    + " private key (%s) properties must be set. The (%s) property"
+                                    + " must be true as well.",
```

Comment:
Here as well, I think the `tlsEnabled` (`tlsRequired`) condition needs to be treated separately from the TLS auth cert/key configuration.

I think we just want to enforce that TLS (HTTPS) is used, not necessarily that *TLS client certificate authentication* is used.